tool-tiles.md

```diff
@@ -36,6 +36,16 @@ export const ToolContentUnion = types.union(
 
 We also specify `ToolTileModel`, a general content model which is the base content model used by all tool tiles. This model contains a `content` property of type `ToolContentUnion`. This allows us to access the `content` property of `ToolTileModel` for any tile, cast it to the unique content model for that tile, and then access properties and methods specific to that tile type.
 
+### ToolMetadataModel
+
+Each tile content model can use a metadata model. This is used to store information that is shared across multiple instances of a document. It is also preserved across reloads. 
+
+When a tool is registered it provides a metadata MST "class". When needed the metadata is looked up in a global map from tile id to metadata instance. If a metadata instance is not found, then one is created and added for this tile id. This code is in `tool-types.ts`.
+
+The metadata is provided to the tile content model via a `doPostCreate` action called on the tile content model.
+
```

Comment:
I would add:
```suggestion

The metadata model was introduced at a time when the response to a remote (i.e. firebase) content change was to replace the document content with a brand new `DocumentContent` instance, which meant that all local state associated with a content instance was regularly lost. We have since moved to use `applySnapshot()` in this situation, which preserves the content instance, so the metadata model should no longer be needed for that purpose.

The metadata model is also used to communicate tile-specific information useful to the tile, notably the tile's id and title. The title is now accessible via the `getToolTileModel()` function and the id could be stored in a volatile property and set with a simple `setTileId()` action.

In short, the notion of tile metadata may have outlived its usefulness and we should endeavor to replace its use with more appropriate mechanisms where possible.

```